### PR TITLE
Fix Planning UI bugs

### PR DESCRIPTION
**Pull request by duanxianbo**
image

image

image

image

[(task:21304)](https://staging.teamyapp.com/teams/107/tasks/21304)

### CODE REVIEW

**Comment by duanxianbo** [comment]
the last commit should fix the updatePhaseModal, now it does not show phase info after the first update, but somehow it still does not work, but i think the commit is already a better version than before

File: src/components/pages/Planning/Project.component.tsx [hunk 1]
```diff
@@ -231,10 +243,30 @@ export function ProjectComponent(props: Props) {
...
+                expectedEndAt: phase.expectedEndAt,
+                status: phase.status,
+            },
+        });
         updatePhaseModalComponentRef.current?.open();
     };
 
+    const onPhaseDetailHandler = (phaseId: number) => (event: MouseEvent) => {
+        event.stopPropagation();
+
```
Comment (naturetime10) [inline src/components/pages/Planning/Project.component.tsx]:
```suggestion
```

File: src/components/pages/Planning/Project.component.tsx [hunk 2]
```diff
@@ -260,21 +291,38 @@ export function ProjectComponent(props: Props) {
...
+                            <MaterialIconUI>edit</MaterialIconUI>
+                        </div>
+                    </TooltipUI>
+                    <TooltipUI
+                        message='Details'
+                        relativeLayout={props.deps.relativeLayout}
+                    >
+                        <div
+                            className={styles.DetailsIcon}
+                            onClick={onPhaseDetailHandler(phase.id)}
```
Comment (naturetime10) [inline src/components/pages/Planning/Project.component.tsx]:
```suggestion
                            onClick={onPhaseDetailClickHandler(phase.id)}
```

File: src/components/pages/Planning/Project.component.tsx [hunk 3]
```diff
@@ -231,10 +243,30 @@ export function ProjectComponent(props: Props) {
...
+                name: phase.name,
+                expectedStartAt: phase.expectedStartAt,
+                expectedEndAt: phase.expectedEndAt,
+                status: phase.status,
+            },
+        });
         updatePhaseModalComponentRef.current?.open();
     };
 
+    const onPhaseDetailHandler = (phaseId: number) => (event: MouseEvent) => {
```
Comment (naturetime10) [inline src/components/pages/Planning/Project.component.tsx]:
```suggestion
    const onPhaseDetailClickHandler = (phaseId: number) => (event: MouseEvent) => {
```

File: src/components/pages/Planning/StoryTable.component.tsx [hunk 4]
```diff
@@ -286,7 +316,7 @@ export const StoryTable: FC<Props> = (props) => {
                         </td>
                         <td>
                             <div className={styles.StoryStatus}>
-                                <StoryStatus status={story.status} />
+                                <StoryStatus relativeLayout={props.deps.relativeLayout} status={story.status} onStatusChange={handleStoryStatusChanged(story)}/>
```
Comment (naturetime10) [inline src/components/pages/Planning/StoryTable.component.tsx]:
```suggestion
                                <StoryStatus 
                                    relativeLayout={props.deps.relativeLayout} 
                                    status={story.status} 
                                    onStatusChange={handleStoryStatusChanged(story)}/>
```

**Review by naturetime10 — APPROVED** [review]

**Pull request merged**